addon/services/engine-router-service.js

```diff
@@ -0,0 +1,121 @@
+import Ember from 'ember';
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import { computed } from '@ember/object';
+import { reads } from '@ember/object/computed';
+import { getOwner } from '@ember/application';
+
+// @HACK: since `getEngineParent` is not exported
+function getEngineParent(engine) {
+  const symbolPrefix = `__ENGINE_PARENT${Ember.GUID_KEY}`;
+  const symbol = Object.keys(engine).find(k => k.startsWith(symbolPrefix));
+  if (!symbol) {
+    return null;
+  }
+  return engine[symbol];
+}
+
+export default Service.extend({
+  engine: computed(function () {
+    return getOwner(this);
+  }),
+
+  externalRoutes: reads('engine._externalRoutes'),
```

Comment:
should be underscored, also these are immutable so we should grab them during `init` and store them (not use a CP)